File: android/lib/net/request/filter_stores_request.dart
import 'package:android/model/filters.dart';
import 'package:android/net/request/request.dart';
import 'package:json_annotation/json_annotation.dart';

part 'filter_stores_request.g.dart';

@JsonSerializable()
class FilterStoresRequest extends Request {
  @JsonKey(name: "Filters")
  final Filters c_filters;

  FilterStoresRequest(this.c_filters) : super(Action.filterStores);

  Filters get filters => c_filters;

  factory FilterStoresRequest.fromJson(Map<String, dynamic> json) =>
      _$FilterStoresRequestFromJson(json);
  Map<String, dynamic> toJson() => _$FilterStoresRequestToJson(this);
}

File: app/src/main/java/com/github/giorgosathanasopoulos/distributed_systems_aueb/android/backend/network/request/FilterStoresRequest.java
package com.github.giorgosathanasopoulos.distributed_systems_aueb.android.backend.network.request;

import com.google.gson.annotations.SerializedName;

import com.github.giorgosathanasopoulos.distributed_systems_aueb.android.backend.json.JsonUtils;
import com.github.giorgosathanasopoulos.distributed_systems_aueb.android.backend.model.Filters;

public class FilterStoresRequest extends Request {
    @SerializedName("Filters")
    private final Filters c_Filters;

    public FilterStoresRequest(UserAgent p_UserAgent, int p_Id, Filters p_Filters) {
        super(p_UserAgent, p_Id, Action.FILTER_STORES);
        this.c_Filters = p_Filters;
        setSrc(JsonUtils.toJson(this));
    }

    public Filters getFilters() {
        return c_Filters;
    }
}
